core/include/scipp/core/string.h

```diff
@@ -52,14 +53,20 @@ std::string to_string(const MutableView<T, U> &mutableView) {
 }
 
 template <class T> std::string array_to_string(const T &arr);
+template <class T> std::string array_to_string(const T &arr, const std::string &unit);
 
-template <class T> std::string element_to_string(const T &item) {
+const std::string to_iso_date(const scipp::core::time_point &tem, const std::string &unit);
+
+template <class T> std::string element_to_string(const T &item, const std::string &unit) {
   using std::to_string;
   if constexpr (std::is_same_v<T, std::string>)
     return {'"' + item + "\", "};
   else if constexpr (std::is_same_v<T, bool>)
     return core::to_string(item) + ", ";
-  else if constexpr (std::is_same_v<T, Eigen::Vector3d>)
+
+  else if constexpr (std::is_same_v<T, scipp::core::time_point>) {
+    return core::to_string(to_iso_date(item, unit)) + ", ";
```

Comment:
I can't find the place that inserts the unit in this case? It should happen somewhere in `to_string` for `Variable` I guess, can you point me to where it is done?

Reply:
Scratch that, found it below.